### PR TITLE
Adds authentication from Github using Azures B2C tenant system. Allows user login and account information retrieval

### LINKED ISSUE #125: As a developer I would like to choose and add a login functionality to the Chirp! application
The app needs a login functionality. Choose one below and integrate it into the app.

- [ ] ASP.NET Core Identity
. Use Individual Accounts and subsequently enable GitHub using [External OAuth authentication 

- [x] Azure AD B2C
. Use Microsoft Identity Platform to target your Azure AD B2C tenant and link it up to GitHub
- [ ] Azure Web Apps Easy Auth
. Configure Easy Auth on your Azure Web App with GitHub and ensure public pages are public while other pages are prompted for login.

### PR DESCRIPTION
**Pull request by Slug-Boi**
### Adds Login and logout features to the Razor page using Azures B2C tenant system
This allows the user to login using github and signout from the site. This was implemented using the Azure B2C tenant system with userflows, app registrations and the identity provider being github. The site will redirect the user to githubs login and authentication and afterwards will return the user to the chirp site now being logged in

### User information retrieval
This new feature also allows for easy and seamless user information retrieval as when the user logs in using github it will send a token back to the program with the users username which can then be accessed and displayed on the site. This will also be useful later for automatically populating fields like email, name and so on if account information becomes a requirement later.

Resolves #125

### CODE REVIEW

**Review by Alex-itu — CHANGES_REQUESTED** [review]
Before merging, I would like to have the conflicts resolved.

**Comment by Slug-Boi** [comment]
> Before merging, I would like to have the conflicts resolved.

Done

**Review by Alex-itu — APPROVED** [review]
With conflicts resolved and after having reviewed the request contents, I Approve this.

File: src/Chirp.Web/Chirp.Web.csproj [hunk 1]
```diff
@@ -23,7 +24,13 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="7.0.10" />
```
Comment (Haruyzal) [inline src/Chirp.Web/Chirp.Web.csproj:27]: We already have `Microsoft.VisualStudio.Web.CodeGeneration.Design version 7.0.11`, so unless the latest version is unstable, I'd recommend removing `7.0.10`, as the potential downgrade is also noted as an error :)

**Review by Haruyzal — CHANGES_REQUESTED** [review]
Well implemented, found it very readable. When the duplicate .csproj package-ref has been taken cared of, I'd def approve :)

File: src/Chirp.Web/Pages/_ViewImports.cshtml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿@using Chirp
+@namespace Chirp.Pages
```
Comment (Haruyzal) [inline src/Chirp.Web/Pages/_ViewImports.cshtml:2]: I presume that the namespace is missing web such that `Chirp.Web.Pages` ? :)

File: src/Chirp.Web/Pages/_ViewStart.cshtml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+﻿@{
```
Comment (Haruyzal) [inline src/Chirp.Web/Pages/_ViewStart.cshtml:1]: Should the `_ViewStart`-file also be housed in `namespace Chirp.Web.Pages` ? o:

**Review by Haruyzal — CHANGES_REQUESTED** [review]
Namespaces! To prevent us from getting minor headaches over such in the future :)

File: src/Chirp.Web/Chirp.Web.csproj (hunk 1 above)
Comment (Slug-Boi) [reply src/Chirp.Web/Chirp.Web.csproj]: Done

File: src/Chirp.Web/Pages/_ViewImports.cshtml (hunk 2 above)
Comment (Slug-Boi) [reply src/Chirp.Web/Pages/_ViewImports.cshtml:2]: This is code copied from the template and seems to work fine as when i comment it out it breaks the login button so I wouldn't worry about it

File: src/Chirp.Web/Pages/_ViewStart.cshtml (hunk 3 above)
Comment (Slug-Boi) [reply src/Chirp.Web/Pages/_ViewStart.cshtml:1]: This is code copied from the template and seems to work fine as when i comment it out it breaks the login button so I wouldn't worry about it

**Review by August-Brandt — APPROVED** [review]
This all looks good to merge

**Review by JonasSkjodt — APPROVED** [review]
Looks good, well done with the code.

There are some _using_'s we dont't use anymore in chirp.web/program.cs like using Microsoft.AspNetCore.Authentication and others, but those are the only minor things i could find.

**Pull request merged**